src/cpu/x64/jit_uni_eltwise_injector.cpp

```diff
@@ -1228,106 +1227,32 @@ void jit_uni_eltwise_injector_f32<isa>::round_compute_vector_fwd(
-        h->vcvtps2dq(vmm_aux1, vmm_aux1);
-    } else {
-        h->uni_vcvtps2dq(vmm_aux1, vmm_src);
-        h->uni_vpsignd(vmm_aux1, vmm_aux1, table_val(mish, 27));
-    }
-
-    h->uni_vpaddd(vmm_aux1, vmm_aux1, table_val(mish, 4));
-    h->uni_vpslld(vmm_aux1, vmm_aux1, 23); //vmm_aux1 = 2^-fx
-    // calculate ln(1 + y)
-    h->uni_vaddps(vmm_aux3, vmm_aux3, vmm_aux1);
-    // x = y; y is free; keep x for further computations
-    h->uni_vmovups(vmm_src, vmm_aux3);
-    // frexp()
-    h->uni_vpsrld(vmm_src, vmm_src, 23);
-    h->uni_vcvtdq2ps(vmm_src, vmm_src);
-    // got n. where n is x = 2^n * y. y = 0.5 .. 1
-    h->uni_vsubps(vmm_src, vmm_src, table_val(mish, 28));
-
-    h->uni_vandps(vmm_aux3, vmm_aux3, table_val(mish, 29));
-    // got y. (mantisa)  0.5 < y < 1
-    h->uni_vorps(vmm_aux3, vmm_aux3, table_val(mish, 30));
-    // y  = y - 1
-    h->uni_vsubps(vmm_aux3, vmm_aux3, table_val(mish, 0));
-    // y = p8
-    h->uni_vmovups(vmm_aux1, table_val(mish, 39));
-    // y = y * x + p7
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 38));
-    // y = y * x + p6
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 37));
-    // y = y * x + p5
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 36));
-    // y = y * x + p4
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 35));
-    // y = y * x + p3
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 34));
-    // y = y * x + p2
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 33));
-    // y = y * x + p1
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 32));
-    // y = y * x + p0 ; p0 = 0
-    h->uni_vfmadd213ps(vmm_aux1, vmm_aux3, table_val(mish, 31));
-    //calculate ln(2) * n
-    h->uni_vmulps(vmm_src, vmm_src, table_val(mish, 3));
-    h->uni_vaddps(vmm_aux1, vmm_aux1, vmm_src);
-    h->uni_vaddps(vmm_aux1, vmm_aux1, vmm_aux0);
+    // v = (exp(x) + 2.0f) * exp(x)
+    exp_compute_vector_fwd(vmm_src);
+    h->uni_vmovups(vmm_aux0, vmm_src);
+    h->uni_vsubps(vmm_aux0, vmm_aux0, table_val(minus_two));
+    h->uni_vmulps(vmm_src, vmm_src, vmm_aux0);
 
-    // get vmm_mask = src > max logf
-    h->uni_vmovups(vmm_mask, vmm_aux2);
-    // y = (x < max log f) ? soft_relu(x) : x
-    compute_cmp_mask(vmm_mask, table_val(mish, 25), _cmp_gt_os);
-    blend_with_mask(vmm_aux1, vmm_aux2);
-    h->uni_vmovups(vmm_src, vmm_aux1);
+    // v / (v + 2.0f)
+    h->uni_vmovups(vmm_aux0, vmm_src);
+    h->uni_vsubps(vmm_aux0, vmm_aux0, table_val(minus_two));
+    h->uni_vdivps(vmm_src, vmm_src, vmm_aux0);
 
-    // tanh(ln(1+exp(x)))
-    tanh_compute_vector_fwd(vmm_src);
-    // x*tanh(ln(1+exp(x)))
+    // x * v / (v + 2.0f)
     h->uni_vmovups(vmm_aux0, h->ptr[h->rsp]);
     h->add(h->rsp, vlen);
     h->uni_vmulps(vmm_src, vmm_src, vmm_aux0);
+
+    // return linear function for large numbers
+    compute_cmp_mask(vmm_aux0, table_val(mish), _cmp_gt_os);
+    blend_with_mask(vmm_src, vmm_aux0);
```

Comment:
I'm just wondering if we could check the whole vector is greater than 20, would it make sense to skip the computations at all and just return the input values?

Reply:
Mish is activation function and so I think this case is not often. Moreover It may will complicate the logic but perf are not obvious